src/fondant/pipeline.py

```diff
@@ -153,6 +155,132 @@ def from_registry(
             ephemeral_storage_size=ephemeral_storage_size,
         )
 
+    @staticmethod
+    def get_image_manifest(image_ref: str):
+        """Retrieve the Docker image manifest.
+
+        Args:
+            image_ref: The Docker image reference (e.g., 'registry/image:tag').
+
+        Returns:
+            dict: The parsed JSON manifest.
+        """
+        cmd = ["docker", "manifest", "inspect", image_ref]
+        try:
+            result = subprocess.run(  # nosec
+                cmd,
+                capture_output=True,
+                check=True,
+                text=True,
+            )
+            return json.loads(result.stdout)
+        except subprocess.CalledProcessError as e:
+            msg = f"Error executing command: {e}"
+            raise Exception(msg)
+        except json.JSONDecodeError as e:
+            msg = f"Error decoding response: {e}"
+            raise Exception(msg)
+
+    def get_component_image_hash(self, image_ref):
+        """Calculate the hash of a Docker image reference. If multiple builds exist for
+        different operating systems, the hash of the amd64 CPU architecture and the linux
+        operating system is returned.
+
+        Args:
+            image_ref (str): The Docker image reference (e.g., 'registry/image:tag').
+
+        Returns:
+            str: The hash value (digest) of the Docker image.
+        """
+        manifest = self.get_image_manifest(image_ref)
```

Comment:
I guess getting the digest of local images is easy, it is the remote part that makes this difficult

Reply:
yeah It should be easy but we still have to somehow build the image -> get the digest -> compile
or we do it at runtime in which case we have to figure out how to pass the digest to the image itself. I checked a bit and it's not available in the image itself unless passed explicitly as an environmental variable. It's a bit of a chicken and an egg problem.